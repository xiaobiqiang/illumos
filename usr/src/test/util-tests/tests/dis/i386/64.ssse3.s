/*
 * This file and its contents are supplied under the terms of the
 * Common Development and Distribution License ("CDDL"), version 1.0.
 * You may only use this file in accordance with the terms of version
 * 1.0 of the CDDL.
 *
 * A full copy of the text of the CDDL should have accompanied this
 * source.  A copy of the CDDL is also available via the Internet at
 * http://www.illumos.org/license/CDDL.
 */

/*
 * Copyright 2016 Joyent, Inc.
 */

/*
 * Test SSSE3 related instructions
 */

.text
.align 16
.globl libdis_test
.type libdis_test, @function
libdis_test:
	pabsb		%mm0, %mm1
	pabsb		(%rax), %mm1
	pabsb		%xmm0, %xmm1
	pabsb		(%rax), %xmm1
	pabsd		%mm0, %mm1
	pabsd		(%rax), %mm1
	pabsd		%xmm0, %xmm1
	pabsd		(%rax), %xmm1
	pabsw		%mm0, %mm1
	pabsw		(%rax), %mm1
	pabsw		%xmm0, %xmm1
	pabsw		(%rax), %xmm1
	palignr		$0x23, %mm0, %mm1
	palignr		$0x23, (%rax), %mm1
	palignr		$0x23, %xmm0, %xmm1
	palignr		$0x23, (%rax), %xmm1
	phaddd		%mm0, %mm1
	phaddd		(%rax), %mm1
	phaddd		%xmm0, %xmm1
	phaddd		(%rax), %xmm1
	phaddw		%mm0, %mm1
	phaddw		(%rax), %mm1
	phaddw		%xmm0, %xmm1
	phaddw		(%rax), %xmm1
	phaddsw		%mm0, %mm1
	phaddsw		(%rax), %mm1
	phaddsw		%xmm0, %xmm1
	phaddsw		(%rax), %xmm1
	phsubd		%mm0, %mm1
	phsubd		(%rax), %mm1
	phsubd		%xmm0, %xmm1
	phsubd		(%rax), %xmm1
	phsubw		%mm0, %mm1
	phsubw		(%rax), %mm1
	phsubw		%xmm0, %xmm1
	phsubw		(%rax), %xmm1
	phsubsw		%mm0, %mm1
	phsubsw		(%rax), %mm1
	phsubsw		%xmm0, %xmm1
	phsubsw		(%rax), %xmm1
	pmaddubsw	%mm0, %mm1
	pmaddubsw	(%rax), %mm1
	pmaddubsw	%xmm0, %xmm1
	pmaddubsw	(%rax), %xmm1
	pmulhrsw	%mm0, %mm1
	pmulhrsw	(%rax), %mm1
	pmulhrsw	%xmm0, %xmm1
	pmulhrsw	(%rax), %xmm1
	pshufb		%mm0, %mm1
	pshufb		(%rax), %mm1
	pshufb		%xmm0, %xmm1
	pshufb		(%rax), %xmm1
	psignb		%mm0, %mm1
	psignb		(%rax), %mm1
	psignb		%xmm0, %xmm1
	psignb		(%rax), %xmm1
	psignd		%mm0, %mm1
	psignd		(%rax), %mm1
	psignd		%xmm0, %xmm1
	psignd		(%rax), %xmm1
	psignw		%mm0, %mm1
	psignw		(%rax), %mm1
	psignw		%xmm0, %xmm1
	psignw		(%rax), %xmm1
.size libdis_test, [.-libdis_test]
