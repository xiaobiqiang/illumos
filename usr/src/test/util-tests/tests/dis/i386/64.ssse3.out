    libdis_test:       0f 38 1c c8        pabsb  %mm0,%mm1
    libdis_test+0x4:   0f 38 1c 08        pabsb  (%rax),%mm1
    libdis_test+0x8:   66 0f 38 1c c8     pabsb  %xmm0,%xmm1
    libdis_test+0xd:   66 0f 38 1c 08     pabsb  (%rax),%xmm1
    libdis_test+0x12:  0f 38 1e c8        pabsd  %mm0,%mm1
    libdis_test+0x16:  0f 38 1e 08        pabsd  (%rax),%mm1
    libdis_test+0x1a:  66 0f 38 1e c8     pabsd  %xmm0,%xmm1
    libdis_test+0x1f:  66 0f 38 1e 08     pabsd  (%rax),%xmm1
    libdis_test+0x24:  0f 38 1d c8        pabsw  %mm0,%mm1
    libdis_test+0x28:  0f 38 1d 08        pabsw  (%rax),%mm1
    libdis_test+0x2c:  66 0f 38 1d c8     pabsw  %xmm0,%xmm1
    libdis_test+0x31:  66 0f 38 1d 08     pabsw  (%rax),%xmm1
    libdis_test+0x36:  0f 3a 0f c8 23     palignr $0x23,%mm0,%mm1
    libdis_test+0x3b:  0f 3a 0f 08 23     palignr $0x23,(%rax),%mm1
    libdis_test+0x40:  66 0f 3a 0f c8 23  palignr $0x23,%xmm0,%xmm1
    libdis_test+0x46:  66 0f 3a 0f 08 23  palignr $0x23,(%rax),%xmm1
    libdis_test+0x4c:  0f 38 02 c8        phaddd %mm0,%mm1
    libdis_test+0x50:  0f 38 02 08        phaddd (%rax),%mm1
    libdis_test+0x54:  66 0f 38 02 c8     phaddd %xmm0,%xmm1
    libdis_test+0x59:  66 0f 38 02 08     phaddd (%rax),%xmm1
    libdis_test+0x5e:  0f 38 01 c8        phaddw %mm0,%mm1
    libdis_test+0x62:  0f 38 01 08        phaddw (%rax),%mm1
    libdis_test+0x66:  66 0f 38 01 c8     phaddw %xmm0,%xmm1
    libdis_test+0x6b:  66 0f 38 01 08     phaddw (%rax),%xmm1
    libdis_test+0x70:  0f 38 03 c8        phaddsw %mm0,%mm1
    libdis_test+0x74:  0f 38 03 08        phaddsw (%rax),%mm1
    libdis_test+0x78:  66 0f 38 03 c8     phaddsw %xmm0,%xmm1
    libdis_test+0x7d:  66 0f 38 03 08     phaddsw (%rax),%xmm1
    libdis_test+0x82:  0f 38 06 c8        phsubd %mm0,%mm1
    libdis_test+0x86:  0f 38 06 08        phsubd (%rax),%mm1
    libdis_test+0x8a:  66 0f 38 06 c8     phsubd %xmm0,%xmm1
    libdis_test+0x8f:  66 0f 38 06 08     phsubd (%rax),%xmm1
    libdis_test+0x94:  0f 38 05 c8        phsubw %mm0,%mm1
    libdis_test+0x98:  0f 38 05 08        phsubw (%rax),%mm1
    libdis_test+0x9c:  66 0f 38 05 c8     phsubw %xmm0,%xmm1
    libdis_test+0xa1:  66 0f 38 05 08     phsubw (%rax),%xmm1
    libdis_test+0xa6:  0f 38 07 c8        phsubsw %mm0,%mm1
    libdis_test+0xaa:  0f 38 07 08        phsubsw (%rax),%mm1
    libdis_test+0xae:  66 0f 38 07 c8     phsubsw %xmm0,%xmm1
    libdis_test+0xb3:  66 0f 38 07 08     phsubsw (%rax),%xmm1
    libdis_test+0xb8:  0f 38 04 c8        pmaddubsw %mm0,%mm1
    libdis_test+0xbc:  0f 38 04 08        pmaddubsw (%rax),%mm1
    libdis_test+0xc0:  66 0f 38 04 c8     pmaddubsw %xmm0,%xmm1
    libdis_test+0xc5:  66 0f 38 04 08     pmaddubsw (%rax),%xmm1
    libdis_test+0xca:  0f 38 0b c8        pmulhrsw %mm0,%mm1
    libdis_test+0xce:  0f 38 0b 08        pmulhrsw (%rax),%mm1
    libdis_test+0xd2:  66 0f 38 0b c8     pmulhrsw %xmm0,%xmm1
    libdis_test+0xd7:  66 0f 38 0b 08     pmulhrsw (%rax),%xmm1
    libdis_test+0xdc:  0f 38 00 c8        pshufb %mm0,%mm1
    libdis_test+0xe0:  0f 38 00 08        pshufb (%rax),%mm1
    libdis_test+0xe4:  66 0f 38 00 c8     pshufb %xmm0,%xmm1
    libdis_test+0xe9:  66 0f 38 00 08     pshufb (%rax),%xmm1
    libdis_test+0xee:  0f 38 08 c8        psignb %mm0,%mm1
    libdis_test+0xf2:  0f 38 08 08        psignb (%rax),%mm1
    libdis_test+0xf6:  66 0f 38 08 c8     psignb %xmm0,%xmm1
    libdis_test+0xfb:  66 0f 38 08 08     psignb (%rax),%xmm1
    libdis_test+0x100: 0f 38 0a c8        psignd %mm0,%mm1
    libdis_test+0x104: 0f 38 0a 08        psignd (%rax),%mm1
    libdis_test+0x108: 66 0f 38 0a c8     psignd %xmm0,%xmm1
    libdis_test+0x10d: 66 0f 38 0a 08     psignd (%rax),%xmm1
    libdis_test+0x112: 0f 38 09 c8        psignw %mm0,%mm1
    libdis_test+0x116: 0f 38 09 08        psignw (%rax),%mm1
    libdis_test+0x11a: 66 0f 38 09 c8     psignw %xmm0,%xmm1
    libdis_test+0x11f: 66 0f 38 09 08     psignw (%rax),%xmm1
